src/braket/default_simulator/simulator.py

```diff
@@ -407,7 +407,21 @@ def _formatted_measurements(
         ]
         #  Gets the subset of measurements from the full measurements
         if measured_qubits is not None and measured_qubits != []:
-            measurements = np.array(measurements)[:, measured_qubits].tolist()
+            if any(qubit in range(simulation.qubit_count) for qubit in measured_qubits):
+                measured_qubits_in_circuit = [
+                    qubit for qubit in measured_qubits if qubit < simulation.qubit_count
+                ]
+                measured_qubits_not_in_circuit = [
+                    qubit for qubit in measured_qubits if qubit >= simulation.qubit_count
+                ]
+                measurements = np.array(measurements)[:, measured_qubits_in_circuit].tolist()
+                measurements = np.pad(
+                    measurements, ((0, 0), (0, len(measured_qubits_not_in_circuit)))
+                ).tolist()
```

Comment:
I find the conversions between lists and numpy pretty confusing. 

Is there a way to keep them as numpy arrays? Maybe somethign like this:

```    
measurements_array = np.array(measurements)
selected_measurements = measurements_array[:, measured_qubits_in_circuit]


measurements = np.pad(selected_measurements, ((0, 0), (0, len(measured_qubits_not_in_circuit)))).tolist()
```

Reply:
Updated to keep `measurements_array` and `selected_measurements` as a numpy array.